client/protocols/export.py

```diff
@@ -0,0 +1,125 @@
+from client.protocols.common import models
+from client.utils import auth
+from client.utils import network
+from datetime import datetime
+from urllib import error
+import client
+import logging
+import os
+import pickle
+
+log = logging.getLogger(__name__)
+
+GAE_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+class ExportProtocol(models.Protocol):
+    """Downloads submissions from the server"""
+
+    def run(self, messages):
+        """If the --export parameter was used, downloads all submissions 
+        for the current assignment from the server and then exits
+        """
+        data = None
+        try:
+            data = pickle.load(open("export_cache.pkl", "rb"))
+            self.access_token = auth.authenticate(self.args.authenticate)
+            log.info('Authenticated with access token %s', self.access_token)
+        except IOError:
+            if self.args.export:
+                self.access_token = auth.authenticate(self.args.authenticate)
+                log.info('Authenticated with access token %s', self.access_token)
+                data = {
+                    'endpoint': self.assignment.endpoint   
+                }
+                data['assign'], data['course'] = self.get_ids(data['endpoint'])
+                data['students'] = self.get_students(data['course'])
+                if not data['students']:
+                    return
+                data['current'] = 0
+        
+        if data:
+            i = 0
+            try:
+                if not os.path.exists('submissions'): 
+                    os.makedirs('submissions')
+                
+                downloads_left = len(data['students']) - data['current']
+                
+                print("{0} submissions left to download".format(downloads_left))
+                
+                for i in range(data['current'], len(data['students'])):
+                    student = data['students'][i]
+                    try:
+                        self.download_submission(student, data['assign'])
+                        data['current'] = i + 1
```

Comment:
This line should occur after the `try/except`, since there's no way it could trigger an `IOError` or an `HTTPError`.